internal/pkg/api/repository/common/metrics.go

```diff
@@ -52,5 +48,12 @@ func CalculateMetrics(executionsMetrics []testkube.ExecutionsMetricsExecutions)
 	metrics.ExecutionDurationP99ms = int32(durationP99 / time.Millisecond)
 
 	return
+}
 
+func calculate(durations []float64, quantile float64) float64 {
+	percentile, err := stats.PercentileNearestRank(durations, quantile)
+	if err != nil {
+		return 0
```

Comment:
log error just in case something unexpected happen. good job!

Reply:
fixed